src/java/org/apache/cassandra/service/CassandraDaemon.java

```diff
@@ -224,12 +224,12 @@ protected void setup()
         // This should be the first write to SystemKeyspace (CASSANDRA-11742)
         SystemKeyspace.persistLocalMetadata();
 
-        maybeInitJmx();
-
         Directories.scheduleVerifyingDiskDoesNotExceedThresholdChecks();
 
         doNotStartupClientInterfacesIfDisabled();
-        completeSetupMayThrowSstableException();
+        beginSetupMayThrowSstableException();
+        maybeInitJmx();
```

Comment:
Can you add a small comment on why it's important to init jmx after the previous function? Theoretically if we get rid of product cleanup this won't be an issue anymore but just in case